### ISSUE TITLE
"Parameter implicitly has an 'any' type." error when passing action-wrapped function with implicit argument types to optional property

### ISSUE THREAD

**Issue by seanchas116** [mobxjs/mobx#2670]
**Intended outcome:**

`action` -wrapped function with implicit argument types can be assigned as an optional property of a object, and the types are correctly inferred.

```ts
import { action } from "mobx";

// Both should be OK

const tes1: {
  a: (x: number) => void;
} = {
  a: action((x) => console.log(x.toFixed()))
};

const test2: {
  a?: (x: number) => void;
} = {
  a: action((x) => console.log(x.toFixed()))
};
```

**Actual outcome:**

Implicit argument types in `action` -wrapped function assigned as an optional property of a object becomes `any` and causes an error in struct environments. 

```ts
import { action } from "mobx";

// OK
const tes1: {
  a: (x: number) => void;
} = {
  a: action((x) => console.log(x.toFixed()))
};

// OK in MobX 5, Error in MobX 6 (x becomes any)
const test2: {
  a?: (x: number) => void;
} = {
  a: action((x) => console.log(x.toFixed()))
  // ↑ "Parameter 'x' implicitly has an 'any' type. ts(7006)" error in this line
};
```

Looks like MobX 5 works fine because `undefined` is added to `T` in [this line](https://github.com/mobxjs/mobx/blob/c194a0adf360cc46cae0dbd806a51887fa6f35b2/src/v5/api/action.ts#L14). Adding `undefined` to `T` in Mobx6 `action.d.ts` fixed the problem in my environment.

**How to reproduce the issue:**

https://codesandbox.io/s/mobx-action-error-gnd8v

**Versions**

Occurs in MobX 6. MobX 5 works fine.

**Comment by kubk** [comment]
It's better to add both `undefined` and `null` to support promises: https://github.com/mobxjs/mobx/pull/2213
Mobx 5 also had unit tests for such cases: https://github.com/mobxjs/mobx/pull/2217
But they were removed in Mobx 6

**Comment by kubk** [comment]
Can make a PR to fix it if anyone interested

**Comment by mweststrate** [comment]
Should be fixed in 6.0.5
